VinylSignalQuality in WSpinnyGLSL

m0dB opened the pull request:
transfer the VinylSignalQualityReport scope to a texture with glTexSubImage2D and use a special shader to draw the VinylSignalQuality overlay on WSpinnyGLSL

Fixes https://github.com/mixxxdj/mixxx/issues/11734

m0dB committed:
transfer the VinylSignalQualityReport scope to a texture with glTexSubImage2D and use a special shader to draw the VinylSignalQuality overlay on WSpinnyGLSL

NotYourAverageAl commented:
I can confirm this fixes the issue. Thanks @m0dB

m0dB commented:
Great! Thanks for confirming.

JoergAtGithub commented:
I can confirm this too. I noticed one bug (which also exist in 2.3), if I switch the skin the VinylSignalQuality donut disappears until I toggle the enable VinylControl button.

m0dB commented:
I will have a look to see if that's an easy fix. While at it...

JoergAtGithub commented on src/widget/wspinnybase.cpp:
```diff
@@ -106,6 +107,10 @@ WSpinnyBase::~WSpinnyBase() {
 #endif
 }
 
+bool WSpinnyBase::shouldDrawVinylQuality() const {
+    return m_bVinylActive && m_bSignalActive && m_bDrawVinylSignalQuality;
+}
```
```suggestion
bool WSpinnyBase::shouldDrawVinylQuality() const {
#if defined __VINYLCONTROL__
    return m_bVinylActive && m_bSignalActive && m_bDrawVinylSignalQuality;
#else
    return false;
#endif
}
```
This could simplify the code which calls `shouldDrawVinylQuality()`

m0dB replied:
Are you suggesting to removed the ifdefs there? Makes sense. Actually, this will never become true if __VINYLCONTROL__ is not defined, but I guess it's more clear like this.

ronso0 commented:
> if I switch the skin the VinylSignalQuality donut disappears until I toggle the enable VinylControl button.

I think this is a feature: IIRC the VC processor is idle as long as VC is off, even with input devices configured, so there's no signal to analyze.
I set up VC inputs just for passthrough and I'd like to keep the CPU load low.

JoergAtGithub commented:
The VinylControl button stays in ON state when I switch the skin. Just the donut disappears until I switched it first to OFF and than to ON again. This is clearly a bug an not a feature - but not an important bug of cause.

m0dB committed:
moved ifdef as suggested by @JoergAtGithub

m0dB committed:
initialize viny control state from ControlProxys

m0dB commented:
> The VinylControl button stays in ON state when I switch the skin. Just the donut disappears until I switched it first to OFF and than to ON again. This is clearly a bug an not a feature - but not an important bug of cause.

I agree it's a bug, I think @ronso0 misread your comment. So I went ahead and fixed it. Not an important bug as you say but the fix was very simple.

With that, can this PR be approved and merged so we don't forget about it?

ronso0 commented:
yes, I misread that, sorry for the noise. (should be more thorough when I do review on mobile B)

JoergAtGithub commented:
I just tested this and it worked flawless. I could switch between all waveformtypes and the signal quality donut always appeared when it should. Thank you!
